### DISCUSSION TITLE
Values from a function return

### DISCUSSION THREAD

**Discussion by abdennour** [urfave/cli#1474] [Q&A]
## Checklist

* [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
* [x] Did you perform a search about this feature? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?
Sometimes, Global flag ( used by all subocommands) requires dynamic initialization. Thus, literal default values, values from env vars, values from file content or even values from alt src will not work.
If this dynamic initialization does not done in a one place, it will lead to repeating ourselves then not complying with DRY.

## Solution description

```go
  		Name:    "myflag",
		Value:   "hello",
                ValueFromFunc: () string {
                       // read value set by user ( if any)
                     //.. do something
                    // return a-string
                }
```

## Describe alternatives you've considered

For now, i am just repeating myself everywhere

**Comment by dearchap** [comment]
@abdennour Cant you use a Before func for the command to do the Global Initialization for all the subcommands. The App/Command/Subcommand all have Before funcs, so you can do multi level initialization depending on your needs
